test/integration/controller_binding_test.go

```diff
@@ -0,0 +1,532 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"reflect"
+	"testing"
+
+	// avoid error `servicecatalog/v1beta1 is not enabled`
+	_ "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/install"
+
+	osb "github.com/pmorie/go-open-service-broker-client/v2"
+	fakeosb "github.com/pmorie/go-open-service-broker-client/v2/fake"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	"github.com/kubernetes-incubator/service-catalog/test/util"
+	"net/http"
+)
+
+// TestCreateServiceBindingSuccess successful paths binding
+func TestCreateServiceBindingSuccess(t *testing.T) {
+	cases := []struct {
```

Comment:
I'm guessing that you envision that there will eventually be more test cases for all of these tests. But I suggest that we add the looping and sub-test running when there are actually multiple test cases rather than polluting the tests with extra things to grok from the outset.

Reply:
I won't hold up approval of this PR for this, though, if you have more test cases in flight that will be coming in your next PR.